gamedat.cc

```diff
@@ -480,9 +480,9 @@ void Game_window::write_saveinfo() {
 			else
 				npc = get_npc(party_man->get_member(i - 1));
 
-			char name[18];
+			char name[18] = {0};
 			std::string namestr = npc->get_npc_name();
-			memcpy(name, namestr.c_str(), 18);
+			strcpy(name, namestr.substr(0, 17).c_str());
```

Comment:
This breaks the case of a name with length 18, which is valid and works in the previous version.

Reply:
so name has to be 19 bytes long and everything would work just fine.